go.mod

```diff
@@ -19,25 +17,120 @@ require (
 	gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b // indirect
 	gopkg.in/gcfg.v1 v1.2.0
 	gopkg.in/warnings.v0 v0.1.2
-	k8s.io/api v0.22.2
-	k8s.io/apimachinery v0.22.2
-	k8s.io/apiserver v0.22.0
+	k8s.io/api v0.23.1
+	k8s.io/apimachinery v0.23.1
+	k8s.io/apiserver v0.23.1
 	k8s.io/client-go v9.0.0+incompatible
-	k8s.io/cloud-provider v0.22.0
+	k8s.io/cloud-provider v0.23.1
 	k8s.io/cloud-provider-gcp/providers v0.0.0
-	k8s.io/code-generator v0.22.2
-	k8s.io/component-base v0.22.0
-	k8s.io/component-helpers v0.22.0
-	k8s.io/controller-manager v0.22.0
-	k8s.io/klog/v2 v2.9.0
-	k8s.io/kube-controller-manager v0.22.0
-	k8s.io/kubelet v0.22.0
-	k8s.io/kubernetes v1.22.0
-	k8s.io/metrics v0.22.0
-	k8s.io/utils v0.0.0-20210707171843-4b05e18ac7d9
+	k8s.io/code-generator v0.23.1
+	k8s.io/component-base v0.23.1
+	k8s.io/component-helpers v0.23.1
+	k8s.io/controller-manager v0.23.1
+	k8s.io/klog/v2 v2.30.0
+	k8s.io/kube-controller-manager v0.23.1
+	k8s.io/kubelet v0.23.1
+	k8s.io/kubernetes v1.23.1
+	k8s.io/metrics v0.23.1
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
 	sigs.k8s.io/controller-tools v0.7.0
 )
 
+require (
+	github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1 // indirect
```

Comment:
This seems suspect. Would like to understand why we're pulling this in.

Reply:
```
$ go mod why github.com/Azure/go-ansiterm
# github.com/Azure/go-ansiterm
k8s.io/cloud-provider-gcp/cmd/cloud-controller-manager
k8s.io/cloud-provider/app
k8s.io/component-base/term
github.com/moby/term
github.com/moby/term/windows
github.com/Azure/go-ansiterm
```

Reply:
See also
https://github.com/kubernetes/kubernetes/blob/32f83b2b60a799020f8b512a923410270168ad94/go.mod#L149

Reply:
In case this makes it easier to swallow, this dependency is already in the main branch of this project  https://github.com/kubernetes/cloud-provider-gcp/blob/285a94c070091037a9bc8ef83fbf2a9ee16e4128/go.sum#L22-L25
but the difference here is that the go.mod file has been bumped to Go 1.17, and in 1.17, when Go writes out the go.mod file, it lists first the required explicit dependencies, and then it creates a list of all of the implicit dependencies as well. That's why we are now seeing this as required where we weren't in Go 1.16

Reply:
@bridgetkromhout going to look into why this is an "Azure" project rather than a more generic Microsoft contribution.

Reply:
https://github.com/kubernetes/kubernetes/blob/v1.23.1/staging/src/k8s.io/cloud-provider/app/controllermanager.go#L127-L136

The only usage of `k8s.io/component-base/term` is to figure out how many columns the terminal has, which has the unfortunate side effect of introducing `github.com/moby/term`, `github.com/moby/term/windows` (for windows support), and finally `github.com/Azure/go-ansiterm`